packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewClientHostApiImpl.java

```diff
@@ -87,6 +87,12 @@ public void onReceivedError(
     @Override
     public boolean shouldOverrideUrlLoading(
         @NonNull WebView view, @NonNull WebResourceRequest request) {
+      if (!request.isForMainFrame()) {
```

Comment:
I think your original implementation here was correct. We should still have the callback made to the Dart `WebViewClient`. However, `AndroidWebviewController` should check that the callback is for the main frame before calling `onNavigationRequest`: https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart#L1467

Reply:
I see, I will make that adaptation tomorrow